### PR TITLE
treewide: Support and test against Python 3.11

### PR DESCRIPTION
**Pull request by fliiiix**

### CODE REVIEW

**Comment by fliiiix** [comment]
It might make sense to also upgrade the [Dockerfile](https://github.com/posativ/isso/blob/master/Dockerfile#L29) but i could not tst that so its not included 

```
 => ERROR [isso-builder  3/11] RUN python3 -m venv /isso  && . /isso/bin/activate  && pip3 install --no-cache-dir --upgrade pip  && pip3 install --no-cache-dir gu  0.5s
```

and update the docs build 
https://github.com/posativ/isso/blob/master/.github/workflows/build-upload-docs.yml#L27

same issue here no idea how to test so i did not upgrade it

But I think this PR is ready to go

**Comment by ix5** [comment]
Time indeed moves fast... Totally forgot about this. Would you mind pinging me about this on Wednesday?

**Comment by fliiiix** [comment]
ping

File: CHANGES.rst [hunk 1]
```diff
@@ -21,7 +21,7 @@ Breaking Changes
 Bugfixes & Improvements
 ^^^^^^^^^^^^^^^^^^^^^^^
 
-- TBD
+- Test Python 3.11 support (`#832`_, l33tname)
```
Comment (ix5) [inline CHANGES.rst:24]: Simply "Python 3.11 support"

**Comment by ix5** [comment]
Please amend the slight wording suggestion, and create a separate commit for the `setup.py` change.

Otherwise LGTM

**Review by ix5 — CHANGES_REQUESTED** [review]

**Comment by fliiiix** [comment]
I think i addressed all the feedback let me know if there is something else missing to get this merged

**Comment by ix5** [comment]
Thanks your your contribution and patience!

**Pull request merged**